### PR TITLE
(WIP) for adding a task to create a local kubernetes cluster

### PR DESCRIPTION
**Pull request by lucywyman**
This is a super rought draft, and raises some questions:
1. Should this task bring up the vagrant VMs for you, or is that an exercise for the user? Ideally this will eventually be able to also deploy to cloud providers.
2. I was only able to find examples deploying to VM nodes and not container ones, but can we use Docker instead of Vagrant? For local deployment that seems much easier + lighter
3. Still TODO on this is making the kubeconfig file work with the infra we set up, either Vagrant or Docker.

Please please please leave questions, comments, concerns, advice, improvements, and any other thoughts you might have!

### CODE REVIEW

File: plans/create_docker_cluster.pp [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  TargetSpec $controller,
+  String[1] $k8s_version='1.2.0',
+  String[1] $k8s_provider='vagrant',
+  String[1] $num_nodes = '3'
+) {
+ 
+  # These get automagically set, right?
+  #run_command($controller, "export $PT_os_family='linux'")
+  #run_command($controller, "export $PT_k8s_version=${k8s_version}")
+  run_command($controller, "export $KUBERNETES_PROVIDER=${kubernetes_provider}")
```
Comment (MikaelSmith) [inline plans/create_docker_cluster.pp]: I'm not sure whether this is going to work. `export` only applies within your current shell, and you may get a new shell every time you `run_command`.

Comment (MikaelSmith) [reply plans/create_docker_cluster.pp]: Also, you don't prefix environment variables with `$`.

**Comment by scotty-c** [comment]
@lucywyman there is already a repo where we have this work going on https://github.com/puppetlabs/kream There is also another project inflight that we are building a Kubernetes distro as well

**Comment by scotty-c** [comment]
@lucywyman Kream is not a module it is the development environment for the Kubernetes module which spins up local clusters. As we already have a workflow on local cluster can we try something else on this PR. So we have been looking at writing tasks for the dashboard install, Istio, Helm (that will live in the Helm module) and we have Rook (in the rook module). As most of these deployments are once of tasks.

**Comment by scotty-c** [comment]
closing

**Pull request closed without merging**